mypy_django_plugin/transformers/managers.py

```diff
@@ -104,17 +108,122 @@ def get_funcdef_type(definition: Union[FuncBase, Decorator, None]) -> Optional[P
             # only needed for pluign-generated querysets.
             ret_type = Instance(queryset_info, [manager_instance.args[0], manager_instance.args[0]])
         variables = []
+    args_types = method_type.arg_types[1:]
+    if _has_compatible_type_vars(base_that_has_method):
+        ret_type = _replace_type_var(
+            ret_type, base_that_has_method.defn.type_vars[0].fullname, manager_instance.args[0]
+        )
+        args_types = [
+            _replace_type_var(arg_type, base_that_has_method.defn.type_vars[0].fullname, manager_instance.args[0])
+            for arg_type in args_types
+        ]
 
     # Drop any 'self' argument as our manager is already initialized
     return method_type.copy_modified(
-        arg_types=method_type.arg_types[1:],
+        arg_types=args_types,
         arg_kinds=method_type.arg_kinds[1:],
         arg_names=method_type.arg_names[1:],
         variables=variables,
         ret_type=ret_type,
     )
 
 
+def _get_base_containing_method(queryset_info: TypeInfo, method_name: str) -> TypeInfo | None:
+    """Finding the base that contains the method."""
+    if method_name in queryset_info.names:
+        return queryset_info
+    for mro in queryset_info.mro:
+        for base in mro.bases:
+            if method_name in base.type.names:
+                return base.type
+    return None
+
+
+def _has_compatible_type_vars(type_info: TypeInfo) -> bool:
```

Comment:
Can we yield an error message on incompatibility?

Reply:
we will do that in another PR right ?

Reply:
Yes, sure